java/src/jmri/NamedBean.java

```diff
@@ -132,7 +132,7 @@
 
     /**
      * Get a fully formatted display that includes the SystemName and,
-     * if set, the UserName”.
+     * if set, the UserName.
```

Comment:
It seems that an unicode character has been added lately and NetBeans doesn't like it and is unable to compile the code.

Reply:
Java has supported Unicode characters in code since at least 2004[1], if not earlier, so I doubt that quote is the issue that caused compilation failure. I am unable to get NetBeans to have any complaints about that character; so if your NetBeans installation is complaining about that character and failing to compile, there is something unique in your installation or configuration that should be fixed, since we do use Unicode characters in unit tests, even if we generally avoid them elsewhere (note that Java properties files *do not* support Unicode characters until Java 9, requiring the use of escape sequences in those files).

1: From https://www.oracle.com/technetwork/articles/javase/supplementary-142654.html :
> The Java Language Specification specifies that all Unicode letters and digits can be used in identifiers. Many supplementary characters are letters or digits, and so the Java Language Specification was updated to refer to new code point-based methods to define the legal characters in identifiers. The javac compiler and other tools that need to detect identifiers were changed to use these new methods."

Reply:
I get this error. I have got the error on both Windows 7 and Windows 8.1, in NetBeans 8.2.

```
error: unmappable character for encoding Cp1252
     * if set, the UserNameâ€?.
```

When I open the NetBeans About dialog, I have the line:
`System: Windows 8 version 6.2 running on amd64; Cp1252; sv_SE (nb)`

Reply:
But even if it's something wrong with my installation (which I doubt since I have a clean installation on my Windows 7 computer), the quote seems misplaced and should be removed anyway.

Reply:
Also note that the NetBeans editior shows the unicode character as a quote, but the Java compiler thinks of it as three separate characters.

I use the Oracle Java.

Reply:
JMRI requires UTF-8 for Java source encoding and the NetBeans configuration stored in GitHub for this project defaults to that encoding. Set your copy of NetBeans back to using that, so your editor is in agreement with every other editor. Note that the nature of cross-platform development is that we must use a cross platform character set, not a platform-specific character encoding that we have no reasonable expectation that another platform can use.

For some reason your copy of NetBeans is not using the JMRI netbeans configuration under <root>/nbproject. That does need to be fixed.

Reply:
My concern is not about the quote, it is about you using an editor that mangles the JMRI source code in incorrect and inappropriate ways. You are seeing three characters because the Cp1252 character set is only 8 bytes while the UTF-8 character set is variable length with either 8, 16, 24, or 32 byte characters.

Note that since we have long had Unicode characters in our tests (see https://github.com/JMRI/JMRI/blob/master/java/test/jmri/util/zeroconf/ZeroConfServiceManagerTest.java), and we haven't encountered any reported problems with those characters, there is something in your IDE that must be fixed.

Reply:
Thanks. That test compiles but fails during execution for me.

I have asked for help on the JMRI development list.

Reply:
I did a test changing ZeroConfServiceManagerTest.java and see what happens in Github desktop. It turns out that the NetBeans editor handles the file correct. But the java compiler fails to handle the file correctly.

So the NetBeans editor handles UTF characters as it should. But something goes wrong during compilation.